### PR TITLE
Various easy fixes after running R CMD check

### PR DESCRIPTION
**Pull request by maelle**
@krlmlr I ran R CMD check, fixed the low-hanging fruits, then opened issues for other problems. I'll also add a GHA workflow in another PR, just so we can see the red getting less red (well it won't get less red in practice, but problems will be removed).

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -121,6 +121,9 @@ export(stats_show)
 export(wrap_df)
 export(wrap_integer)
 import(rlang)
+import(vctrs, except = data_frame)
```
Comment (maelle) [inline NAMESPACE:124]: I saw this in dplyr, and here there were a few non-namespaced vctrs functions.

File: R/duckplyr-package.R [hunk 2]
```diff
@@ -65,14 +67,7 @@
...
 #' @importFrom tidyselect everything
 #' @importFrom utils head
-#' @importFrom vctrs list_of
-#' @importFrom vctrs new_data_frame
-#' @importFrom vctrs new_rcrd
-#' @importFrom vctrs unspecified
-#' @importFrom vctrs vec_c
-#' @importFrom vctrs vec_cbind
-#' @importFrom vctrs vec_rbind
-#' @importFrom vctrs vec_slice
```
Comment (krlmlr) [inline R/duckplyr-package.R:75]: I prefer verbose imports here for now, even for the vctrs package. You can leave the alternative syntax as a non-roxygen2 comment.

**Review by krlmlr — COMMENTED** [review]
Thanks, great!

**Comment by krlmlr** [comment]
One thing we could do is to add some files in `.R` to `.Rbuildignore` -- e.g., the `tpch_*()` functions are used only in local operation currently, with `load_all()` .

**Comment by maelle** [comment]
@krlmlr R CMD check is obviously still not passing, but it's a step towards that. :smile_cat:

**Comment by krlmlr** [comment]
Ignore warnings for now?

**Comment by maelle** [comment]
but there's at least #22

**Comment by krlmlr** [comment]
`@noRd` FTW. Small bets 🤓

**Comment by maelle** [comment]
@krlmlr does [`0e52759` (#16)](https://github.com/duckdblabs/duckplyr/pull/16/commits/0e52759923639ac93c2bd221aadb4b4be3e03895) make sense to you?

**Comment by maelle** [comment]
Current test errors

- something about not finding a file. looks related to caching but what?

```r
+   path[1]="/home/runner/.cache/R/R.cache": No such file or directory
```

- `Error in `duckplyr_summarise(.)`: could not find function "duckplyr_summarise"`, coming from a test starting like so

```r
test_that("duckplyr_summarise(.groups=)", {
  expect_message(eval_bare(
    expr(data.frame(x = 1, y = 2) %>% group_by(x, y) %>% duckplyr_summarise()),
    env(global_env())
  ))
```

@krlmlr I'd appreciate any pointer.

**Comment by krlmlr** [comment]
💚 

The r-devel builds are painful without caching, the macOS check seems too slow -- we can disable those at the first sight of any trouble whatsoever.

**Comment by maelle** [comment]
Oh, a new failure

**Pull request merged**